File: client/src/boundary/model/BoundaryBoard.ts
import BoundaryShip from "./BoundaryShip";

export default class BoundaryBoard {
  readonly grid: number[][];
  readonly ships: BoundaryShip[];

  constructor(grid: number[][], ships: BoundaryShip[]) {
    this.grid = grid;
    this.ships = ships;
  }
}

File: server/battleships/use_cases/api/src/main/java/eu/optas/use_cases/api/BoundaryBoard.java
package eu.optas.use_cases.api;

import java.util.List;

public class BoundaryBoard {
    private final List<List<Integer>> grid;
    private final List<BoundaryShip> ships;

    public BoundaryBoard(List<List<Integer>> grid, List<BoundaryShip> ships) {
        this.grid = grid;
        this.ships = ships;
    }

    public List<List<Integer>> getGrid() {
        return grid;
    }

    public List<BoundaryShip> getShips() {
        return ships;
    }
}
